lib/watir-webdriver/elements/select.rb

```diff
@@ -86,17 +86,20 @@ def select_value(str_or_rx)
 
     def selected?(str_or_rx)
       assert_exists
-      matches = element_call do
-        @element.find_elements(:tag_name, 'option').select do |e|
-          str_or_rx === e.text || str_or_rx === e.attribute(:label)
+
+      matches = false
+
+      element_call do
+        @element.find_elements(:tag_name, 'option').each do |e|
+        matched = str_or_rx === e.text || str_or_rx === e.attribute(:label)
+        return true if matched && e.selected?
+        matches = true if matched
         end
       end
 
-      if matches.empty?
-        raise UnknownObjectException, "Unable to locate option matching #{str_or_rx.inspect}"
-      end
+      return false if matches
```

Comment:
Essentially 
if there is a match and it is selected  --> true
if nothing that matches is selected --> false
If nothing matches --> raise error

What backward compatibility does it break?